src/api/app/models/bs_request.rb

```diff
@@ -154,6 +154,9 @@ def self.new_from_xml(xml)
     if hashed['id']
       theid = hashed.delete('id') { raise 'not found' }
       theid = Integer(theid)
+      if BsRequest.find_by_number(theid)
```

Comment:
You validate in the model. When you save the data in the controller you check the return and [handle it](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/configuration_controller.rb#L22).